front/components/members/InvitationModal.tsx

```diff
@@ -0,0 +1,524 @@
+        inviteEmailsList.find(
+          (e) =>
+            m.email === e &&
+            m.workspaces[0].role !== invitationRole &&
+            m.workspaces[0].role !== "none"
+        )
+      ),
+      revoked: members.filter((m) =>
+        inviteEmailsList.find(
+          (e) => m.email === e && m.workspaces[0].role === "none"
+        )
+      ),
+      notInWorkspace: inviteEmailsList.filter(
+        (e) => !members.find((m) => m.email === e)
+      ),
+    };
+
+    const { notInWorkspace, activeDifferentRole, revoked } = invitesByCase;
+
+    const ReinviteUsersMessage = (
+      <div className="mt-6 flex flex-col gap-6 px-2">
+        {revoked.length > 0 && (
+          <div>
+            <div>
+              The user(s) below were previously revoked from your workspace.
+              Reinstating them will also immediately reinstate their
+              conversation history on Dust.
+            </div>
+            <div className="mt-2 flex max-h-48 flex-col gap-1 overflow-y-auto rounded border p-2 text-xs">
+              {revoked.map((user) => (
+                <div
+                  key={user.email}
+                >{`- ${user.fullName} (${user.email})`}</div>
+              ))}
+            </div>
+          </div>
+        )}
+        {activeDifferentRole.length > 0 && (
+          <div>
+            <div>
+              The user(s) below are already in your workspace with a different
+              role. Moving forward will change their role to{" "}
+              <span className="font-bold">{displayRole(invitationRole)}</span>.
+            </div>
+            <div className="mt-2 flex max-h-48 flex-col gap-1 overflow-y-auto rounded border p-2 text-xs">
+              {activeDifferentRole.map((user) => (
+                <div key={user.email}>{`- ${
+                  user.fullName
+                } (current role: ${displayRole(
+                  user.workspaces[0].role
+                )})`}</div>
+              ))}
+            </div>
+          </div>
+        )}
+
+        <div>Do you want to proceed?</div>
+      </div>
+    );
+    const hasExistingMembers =
+      activeDifferentRole.length > 0 || revoked.length > 0;
+
+    const shouldProceedWithInvites =
+      !hasExistingMembers ||
+      (await confirm({
+        title: "Some users are already in the workspace",
+        message: ReinviteUsersMessage,
+        validateLabel: "Yes, proceed",
+        validateVariant: "primaryWarning",
+      }));
+
+    if (shouldProceedWithInvites) {
+      await sendInvitations({
+        owner,
+        emails: notInWorkspace,
+        invitationRole,
+        sendNotification,
+        isNewInvitation: true,
+      });
+
+      if (hasExistingMembers) {
+        await handleMembersRoleChange({
+          members: [...activeDifferentRole, ...revoked],
+          role: invitationRole,
+          sendNotification,
+        });
+        await mutate(`/api/w/${owner.sId}/members`);
```

Comment:
We try to avoid exposing url in the component like this an privilege local mutates of global SWR mutate.

Example:
https://github.com/dust-tt/dust/blob/2e9f9b3c6855ae4ca8ead723b455e3c6a238008a/front/lib/swr.ts#L291-L304